python/HistogramTools.py

```diff
@@ -2,6 +2,36 @@
 
 import ROOT
 import re
+from types import MethodType
+
+class TFileWrapper(ROOT.TFile):
+    """Act as a wrapper around TFile, to make the 'Get' operation faster by keeping an internal dictionary of the file's keys."""
+
+    def __init__(self, *args, **kwargs):
+        ROOT.TFile.__init__(self, *args, **kwargs)
+        self.keys = { key.GetName(): key for key in self.GetListOfKeys() }
+
+    def Get(self, name):
+        key = self.keys.get(name, None)
+        if key:
+            return key.ReadObj()
+        else:
+            raise Exception("Could not read object {} from file.".format(name))
+
+    @staticmethod
+    def GetStatic(self, name):
+        key = self.keys.get(name, None)
+        if key:
+            return key.ReadObj()
+        else:
+            raise Exception("Could not read object {} from file.".format(name))
+    
+    @staticmethod
+    def Open(*args):
+        _file = ROOT.TFile.Open(*args)
+        _file.keys = { key.GetName(): key for key in _file.GetListOfKeys() }
+        _file.Get = MethodType(TFileWrapper.GetStatic, _file)
```

Comment:
I understand python is complaining about that (`MethodType` tries to bind an already bound method in that case), no easy solution for that as far as I know. Ideally you would want a `TFile`->`TFileWrapper` conversion, such that you can return a `TFileWrapper` from `Open`.
If all you use of TFileWrapper are the `Open` and `GetStatic` methods, you could even get rid of the inheritance from `TFile`, the constructor and the non-static `Get` method ;-) . If you just want to avoid code duplication, you can change `Get` into `def Get(self, name): return TFileWrapper.GetStatic(self, name)`